pkg/operator/k8sutil/deployment.go

```diff
@@ -47,12 +51,25 @@ func GetDeploymentSpecImage(clientset kubernetes.Interface, d apps.Deployment, c
 
 // UpdateDeploymentAndWait updates a deployment and waits until it is running to return. It will
 // error if the deployment does not exist to be updated or if it takes too long.
-func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace string) (*v1.Deployment, error) {
+func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace string, clusterInfo *cephconfig.ClusterInfo) (*v1.Deployment, error) {
```

Comment:
The `k8sutil` will be used by other storage providers as well so we should avoid introducing ceph-specific functionality here. What if this method took a "verify()" callback method for checking if it's safe to continue with a daemon? Then the callback could be implemented in the ceph operator for ceph-specific checking.

Reply:
As an example, it might look like this (ignoring syntax errors and incomplete pseudocode). Details like parameters and return vars may need to change as well.
```
func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deployment, namespace string, verifyCallback func(action string) (bool, error)) (*v1.Deployment, error) {
...
   stop, err := verifyCallback("stop")
   if err != nil {
      // return error
   }
   if stop {
     return nil
   } 
}
```
Another method would have a ceph-specific implementation such as:
```
func UpdateCephDeploymentAndWait(...) {

  callback := func(action string) (bool, error) {
    if action == "stop" { 
  	logger.Infof("checking if deployment %s can be updated", deployment.Name)
	err = client.OkToStopOrContinue(context, namespace, deployment.Name, actionToCheck, clusterInfo.CephVersion)
	if err != nil {
		return false, fmt.Errorf("failed to check if deployment %s can be updated: %+v", deployment.Name, err)
	}
    } else if action == "continue" {
       ...
    }
  }
  return UpdateDeploymentAndWait(callback)
```